src/posix/fs.cpp

```diff
@@ -132,6 +149,188 @@ void PosixFileAccess::sysclose()
     }
 }
 
+MUTEX_CLASS PosixFileAccess::asyncmutex(false);
+
+bool PosixFileAccess::asyncavailable()
+{
+    return true;
+}
+
+AsyncIOContext *PosixFileAccess::newasynccontext()
+{
+    return new PosixAsyncIOContext();
+}
+
+void PosixFileAccess::asyncopfinished(union sigval sigev_value)
+{
+    PosixAsyncSynchronizer *synchronizer = (PosixAsyncSynchronizer *)(sigev_value.sival_ptr);
+
+    PosixFileAccess::asyncmutex.lock();
+    PosixAsyncIOContext *context = synchronizer->context;
+    struct aiocb *aiocbp = synchronizer->aiocb;
+
+    if (!context)
+    {
+        LOG_debug << "Async IO request already cancelled";
+        delete synchronizer;
+        delete aiocbp;
+        PosixFileAccess::asyncmutex.unlock();
+        return;
+    }
+
+    context->synchronizer = NULL;
+    context->failed = aio_error(aiocbp);
+    if (!context->failed)
+    {
+        if (context->op == AsyncIOContext::READ)
+        {
+            memset((void *)(((char *)(aiocbp->aio_buf)) + aiocbp->aio_nbytes), 0, context->pad);
+            LOG_verbose << "Async read finished OK";
+        }
+        else
+        {
+            LOG_verbose << "Async write finished OK";
+        }
+    }
+    else
+    {
+        LOG_warn << "Async operation finished with error";
+    }
+
+    delete synchronizer;
+    delete aiocbp;
+
+    context->retry = false;
+    context->finished = true;
+    if (context->userCallback)
+    {
+        context->userCallback(context->userData);
+    }
+    PosixFileAccess::asyncmutex.unlock();
+}
+
+void PosixFileAccess::asyncsysopen(AsyncIOContext *context)
+{
+    string path;
+    path.assign((char *)context->buffer, context->len);
```

Comment:
Yes, it's the path of the file to open. It's not allowed to call it with a NULL path